src/microbit/model/display.py

```diff
@@ -0,0 +1,207 @@
+                )
+            elif isinstance(value, (str, int, float)):
+                if isinstance(value, str):
+                    chars = list(value)
+                else:
+                    chars = list(str(value))
+
+                for c in chars:
+                    self.__image = Display.__get_image_from_char(c)
+                    time.sleep(delay / 1000)
+            else:
+                # Check if iterable
+                try:
+                    _ = iter(value)
+                except TypeError as e:
+                    raise e
+
+                for elem in value:
+                    if isinstance(elem, Image):
+                        self.__image = elem.crop(
+                            0, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
+                        )
+                    elif isinstance(elem, str) and len(elem) == 1:
+                        self.__image = Display.__get_image_from_char(elem)
+                    # If elem is not char or image, break without iterating through rest of list
+                    else:
+                        break
+                    time.sleep(delay / 1000)
+            if not loop:
+                break
+        if clear:
+            self.clear()
+
+    def get_pixel(self, x, y):
+        return self.__image.get_pixel(x, y)
+
+    def set_pixel(self, x, y, value):
+        self.__image.set_pixel(x, y, value)
+
+    def clear(self):
+        self.__image = Image()
+
+    def on(self):
+        self.__on = True
+
+    def off(self):
+        self.__on = False
+
+    def is_on(self):
+        return self.__on
+
+    def read_light_level(self):
+        raise NotImplementedError(CONSTANTS.NOT_IMPLEMENTED_ERROR)
+
+    # Helpers
+
+    def __get_array(self):
+        return self.__image._Image__LED
+
+    def __print(self):
+        print("")
+        for i in range(CONSTANTS.LED_HEIGHT):
+            print(self._Display__image._Image__LED[i])
+
+    @staticmethod
+    def __get_image_from_char(c):
+        # If c is not between the ASCII alphabet we cover, make it a question mark
+        if ord(c) < CONSTANTS.ASCII_START or ord(c) > CONSTANTS.ASCII_END:
+            c = "?"
+        offset = (ord(c) - CONSTANTS.ASCII_START) * 5
+        representative_bytes = CONSTANTS.ALPHABET[offset : offset + 5]
+        return Image(Display.__convert_bytearray_to_image_str(representative_bytes))
+
+    # Removes columns that are not lit
+    @staticmethod
+    def __strip_image(image):
+        min_index = CONSTANTS.LED_WIDTH - 1
+        max_index = 0
+        for row in image._Image__LED:
+            for index, bit in enumerate(row):
+                if bit > 0:
+                    min_index = min(min_index, index)
+                    max_index = max(max_index, index)
+        return image.crop(min_index, 0, max_index - min_index + 1, CONSTANTS.LED_HEIGHT)
+
+    @staticmethod
+    def __convert_bytearray_to_image_str(byte_array):

```

Comment:
maybe add a comment that this "bytearray" isn't the same one as the constructor for the Image, but uses the system that requires you to convert to binary

Reply:
Updated. Let me know what you think